nemoguardrails/library/polygraf/request.py

```
@@ -0,0 +1,93 @@
+    headers: Dict[str, str],
+) -> List[Dict[str, Any]]:
+    async with session.post(server_endpoint, json=payload, headers=headers) as resp:
+        if resp.status != 200:
+            raise ValueError(f"Polygraf call failed with status code {resp.status}.\nDetails: {await resp.text()}")
+
+        try:
+            data = await resp.json()
+        except aiohttp.ContentTypeError:
+            raise ValueError(
+                f"Failed to parse Polygraf response as JSON. Status: {resp.status}, Content: {await resp.text()}"
+            )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)